### ISSUE TITLE
"export_loadings_default_file_format" error with latest master

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1555]
Latest everything.
```python
%matplotlib qt4
import hyperspy.api as hs
```
```bash
---------------------------------------------------------------------------
TraitError                                Traceback (most recent call last)
<ipython-input-2-cb68e1137034> in <module>()
      1 %matplotlib qt4
----> 2 import hyperspy.api as hs

/Users/thomas/miniconda3/envs/flowers/lib/python3.5/site-packages/hyperspy/api.py in <module>()
     73 
     74 from hyperspy.Release import version as __version__
---> 75 from hyperspy import signals
     76 from hyperspy.io import load
     77 from hyperspy.defaults_parser import preferences

/Users/thomas/miniconda3/envs/flowers/lib/python3.5/site-packages/hyperspy/signals.py in <module>()
     40 
     41 # -*- coding: utf-8 -*-
---> 42 from hyperspy.signal import BaseSignal
     43 from hyperspy._signals.lazy import LazySignal
     44 from hyperspy._signals.signal1d import (Signal1D, LazySignal1D)

/Users/thomas/miniconda3/envs/flowers/lib/python3.5/site-packages/hyperspy/signal.py in <module>()
     32 
     33 from hyperspy.axes import AxesManager
---> 34 from hyperspy import io
     35 from hyperspy.drawing import mpl_hie, mpl_hse, mpl_he
     36 from hyperspy.learn.mva import MVA, LearningResults

/Users/thomas/miniconda3/envs/flowers/lib/python3.5/site-packages/hyperspy/io.py in <module>()
     32 from .io_plugins import io_plugins, default_write_ext
     33 from .exceptions import VisibleDeprecationWarning
---> 34 from .defaults_parser import preferences
     35 
     36 _logger = logging.getLogger(__name__)

/Users/thomas/miniconda3/envs/flowers/lib/python3.5/site-packages/hyperspy/defaults_parser.py in <module>()
    311 
    312 # Use the traited classes to cast the content of the ConfigParser
--> 313 config2template(template, config)
    314 
    315 

/Users/thomas/miniconda3/envs/flowers/lib/python3.5/site-packages/hyperspy/defaults_parser.py in config2template(template, config)
    276                 value = float(value)
    277             config_dict[name] = value
--> 278         traited_class.trait_set(True, **config_dict)
    279 
    280 

/Users/thomas/miniconda3/envs/flowers/lib/python3.5/site-packages/traits/has_traits.py in trait_set(self, trait_change_notify, **traits)
   1539         else:
   1540             for name, value in list(traits.items()):
-> 1541                 setattr( self, name, value )
   1542 
   1543         return self

/Users/thomas/miniconda3/envs/flowers/lib/python3.5/site-packages/traits/trait_handlers.py in error(self, object, name, value)
    171         """
    172         raise TraitError( object, name, self.full_info( object, name, value ),
--> 173                           value )
    174 
    175     def full_info ( self, object, name, value ):

TraitError: The 'export_loadings_default_file_format' trait of a MachineLearningConfig instance must be 'msa' or 'hspy' or 'emd' or 'rpl' or 'png' or 'unf' or 'blo' or 'tif', but a value of 'hdf5' <class 'str'> was specified.
```

**Comment by thomasaarholt** [comment]
Right, I believe I've tracked this down to some sort of preferences file that is perhaps not overwritten with a new version of hyperspy? The reason I think this, is that when I print the contents of `name, value` within `defaults_parser.config2template()`, I get the following:
```
preedge_safe_window_width 2.0
eels_gos_files_path /Users/thomas/Google Drive/0_Data/0_TEM Data/0_EELS Results/H-S GOS Tables
fine_structure_smoothing 0.3
fine_structure_active False
synchronize_cl_with_ll False
fine_structure_width 30.0
min_distance_between_edges_for_fine_structure 0.0
default_fitter leastsq
eds_mn_ka 130.0
eds_tilt_stage 0.0
eds_detector_azimuth 0.0
eds_detector_elevation 35.0
multiple_files True
same_window True
export_loadings_default_file_format hdf5
export_factors_default_file_format hdf5
```
Note how "eels_gos_files_path" has remembered where I keep my GOS files, even though I reinstalled hyperspy on a clean conda environment. I'm about to investigate where the preferences file is kept, and I suspect I'll find the "hdf5" default value there.

**Comment by thomasaarholt** [comment]
And that took about 5 seconds. The preferences file on Mac is at:
`/Users/thomas/.hyperspy`

Is this overwritten when we update hyperspy? There may be a chance that I installed hyperspy on my "root" environment once, and therefore it didn't get overwritten with the new version of hyperspy.

**Comment by thomasaarholt** [comment]
If some kind soul would look through the following to see if these settings look correct with what we currently have as default settings in hyperspy (with the exception of the GOS path), I'd appreciate it:
```haskell
[Model]
default_fitter = leastsq

[Plot]
plot_on_load = False
default_style_to_compare_spectra = overlap
pylab_inline = False

[MachineLearning]
same_window = True
export_loadings_default_file_format = hspy
multiple_files = True
export_factors_default_file_format = hspy

[General]
dtb_expand_structures = True
default_file_format = hspy
logger_on = False
interactive = True
show_progressbar = True
default_export_format = rpl
logging_level = WARNING

[EELS]
synchronize_cl_with_ll = False
fine_structure_width = 30.0
eels_gos_files_path = /Users/thomas/Google Drive/0_Data/0_TEM Data/0_EELS Results/H-S GOS Tables
min_distance_between_edges_for_fine_structure = 0.0
preedge_safe_window_width = 2.0
fine_structure_active = False
fine_structure_smoothing = 0.3

[EDS]
eds_mn_ka = 130.0
eds_tilt_stage = 0.0
eds_detector_azimuth = 0.0
eds_detector_elevation = 35.0
```

edit: It looks like ```default_file_format = hdf5``` is a necessary key-pair. Should it be?

**Comment by francisco-dlp** [comment]
Do these settings work for you?

**Comment by thomasaarholt** [comment]
Yes, to clarify: I have fixed the issue by using the (edit-included) settings above.

**Comment by thomasaarholt** [comment]
My concern now is that we need to be aware of this if we change settings in the future.

**Comment by francisco-dlp** [comment]
Sure, this is a bug that we'll need to fix for the next release. The issue is that the default extension for hyperspy's hdf5 is now hspy but the ML export_* methods are not aware of this...

It could be easily fixed, but, I start wondering how sane it is to set certain defaults in preferences, as this makes hyperspy scripts less portable. So, I am for removing certain items from preferences and the export format default are in my mind good candidates...

**Comment by thomasaarholt** [comment]
I am inclined to agree, but we should hear out the rest of the group. I'll create a clearer issue on it.

**Comment by francisco-dlp** [comment]
Thanks!

**Comment by thomasaarholt** [comment]
#1556

**Comment by francisco-dlp** [comment]
This is not yet fixed...

**Comment by francisco-dlp** [comment]
Fixed in #1631
